doc/keynames/README.md

```diff
@@ -0,0 +1,477 @@
+| `/elektra`              | `/elektra/version/info` | "Key 2" is below "Key 1"          |
+| `/elektra/version/info` | `/elektra/version/info` | "Key 1" and "Key 2" are equal     |
+| `/elektra/version/info` | `/elektra/data`         | no relation                       |
+| `/elektra/data`         | `/elektra/version`      | "Key 1" and "Key 2" are siblings  |
+
+You can think of the Key Hierarchy (within a single Namespace) as a big tree of Keys.
+Each node in the tree is a single Key `K` and the children of the node are the Keys that directly below `K`.
+
+![Tree structure of a Key Hierarchy](tree.svg)
+
+The diagram above shows the Key Hierarchy of the Keys in the table above (`A` -> `B` denotes `A` is directly below `B`).
+
+> _Note:_ While we could use the directory vs. file terminology for Elektra as well, it is recommended to avoid it.
+> This is because in Elektra, every Key may have an associated value.
+> In particular a Key may have a value, even if there are other Keys below it.
+> This value is **not**, as a beginner might suspect, the set of Keys below it, like you could say the value of a directory is the set of files (and directories) within.
+> The value is just another value, like any other key would have.
+>
+> Instead, we recommend the terms _leaf Key_ and _non-leaf Key_, as these are commonly used for tree-like structures and their definitions fit perfectly.
+
+#### 1.1.2. The "parent" confusion
+
+As an inverse to "is below" we sometimes use "parent".
+For example:
+
+- `/elektra/version` is directly below `/elektra`, so `/elektra` is the (direct) parent of `/elektra/version`
+- `/elektra/version/info` is below `/elektra`, so `/elektra` is a parent of `/elektra/version`
+- `/elektra/version/info` and `/elektra/version` are below `/elektra`, so `/elektra` is the common parent.
+
+This terminology can be easy to confuse, as "parent" is used for multiple different things.
+If the context doesn't make it clear what "parent" means, you might consider more clear terms, e.g. other common terms for tree-like structures.
+For example, you could use "ancestor" and "direct parent" as a clear differentiation.
+
+There is also the term "(the) Parent Key".
+In many cases, this is refers to a very specific Key in the given context that is a "common parent" to a certain set of Keys.
+If you are working within such a context, be careful about "a parent Key" vs. "the Parent Key".
+
+### 1.2. Namespaces and Root Keys
+
+We mentioned above that there are different Namespaces in Elektra.
+Now we will explain their meaning.
+
+To recap, Elektra knows these Namespaces:
+
+- Cascading
+- Meta
+- Spec
+- Proc
+- Dir
+- User
+- System
+- Default
+
+We mentioned above that there are Key Names with zero Key Name Parts, i.e. just Namespace.
+These are called _Root Keys_ (based on Unix's filesystem root, as well as the root of a tree).
+
+Lets explore them one by one:
```

Comment:
Maybe separate "special" key names from others. Here is some overlap with doc/tutorials/namespaces.md

Reply:
Not sure what you mean by "special" key names... (I assume you are not referring to section 4.1 of this document)

Reply:
Your separation doesn't make sense. Cascading, Meta, Spec stand on their own and Proc, Dir, User, System, Default form a fallback chain. That's the only separation that would make sense IMO, but even that only makes sense **after** the Namespace Resolution section.

Reply:
Yes, it is tricky to get a flow of a document right. For me "default" is special, as it is not an input but derived. So it depends on how exactly the other text around it is.